src/Upgrade/Infrastructure/VersionControlSystem/SourceCodeProvider/GitHub/GitHubClientFactory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace Upgrade\Infrastructure\VersionControlSystem\SourceCodeProvider\GitHub;
+
+use Github\AuthMethod;
+use Github\Client;
+use Github\HttpClient\Builder;
+use Psr\Http\Client\ClientInterface;
+use Upgrade\Infrastructure\Configuration\ConfigurationProvider;
+
+class GitHubClientFactory
+{
+    /**
+     * @var \Github\Client|null
+     */
+    protected ?Client $client = null;
+
+    /**
+     * @var \Upgrade\Infrastructure\Configuration\ConfigurationProvider
+     */
+    protected ConfigurationProvider $configurationProvider;
+
+    /**
+     * @var \Psr\Http\Client\ClientInterface
+     */
+    protected ClientInterface $httpClient;
+
+    /**
+     * @param \Upgrade\Infrastructure\Configuration\ConfigurationProvider $configurationProvider
+     * @param \Psr\Http\Client\ClientInterface $httpClient
+     */
+    public function __construct(ConfigurationProvider $configurationProvider, ClientInterface $httpClient)
+    {
+        $this->configurationProvider = $configurationProvider;
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * @return \Github\Client
+     */
+    public function getClient(): Client
+    {
+        if ($this->client === null) {
+            $this->client = $this->createClient();
+        }
+
+        return $this->client;
+    }
+
+    /**
+     * @return \Github\Client
+     */
+    protected function createClient(): Client
```

Comment:
move this init to constructor

Reply:
That would be unti-pattern. Please, don't. Current logic is fine.